src/Core/Settings.h

```diff
@@ -712,11 +712,21 @@ struct Settings : public BaseSettings<SettingsTraits>, public IHints<2, Settings
     /// (Don't forget to call notify() on the `variables_map` after parsing it!)
     void addProgramOptions(boost::program_options::options_description & options);
 
+    /// Adds program options for clickhouse-format to set the settings from a command line.
+    /// (Don't forget to call notify() on the `variables_map` after parsing it!)
+    void addFormatOptions(boost::program_options::options_description & options);
+
     /// Check that there is no user-level settings at the top level in config.
     /// This is a common source of mistake (user don't know where to write user-level setting).
     static void checkNoSettingNamesAtTopLevel(const Poco::Util::AbstractConfiguration & config, const String & config_path);
 
     std::vector<String> getAllRegisteredNames() const override;
+
+private:
+    void addProgramOption(boost::program_options::options_description & options, const SettingFieldRef & field);
+
+    inline static const std::unordered_set<String> formatSettingNames
```

Comment:
We should not put settings specific to `clickhouse-format` into Core Settings.h, better to implement this logic in Format.cpp.

Reply:
Ok